server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/NodesReloadSecureSettingsRequest.java

```diff
@@ -62,24 +68,60 @@ public ActionRequestValidationException validate() {
         return validationException;
     }
 
-    public String secureSettingsPassword() {
+    public SecureString secureSettingsPassword() {
         return secureSettingsPassword;
     }
 
-    public NodesReloadSecureSettingsRequest secureStorePassword(String secureStorePassword) {
+    public NodesReloadSecureSettingsRequest secureStorePassword(SecureString secureStorePassword) {
         this.secureSettingsPassword = secureStorePassword;
         return this;
     }
 
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
-        secureSettingsPassword = in.readString();
+        final byte[] passwordBytes = in.readByteArray();
+        try {
+            this.secureSettingsPassword = new SecureString(utf8BytesToChars(passwordBytes));
+        } finally {
+            Arrays.fill(passwordBytes, (byte) 0);
+        }
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeString(secureSettingsPassword);
+        final byte[] passwordBytes = toUtf8Bytes(this.secureSettingsPassword.getChars());
+        try {
+            out.writeByteArray(passwordBytes);
+        } finally {
+            Arrays.fill(passwordBytes, (byte) 0);
+        }
+    }
+
+    /**
+     * Encodes the provided char[] to a UTF-8 byte[]. The provided char[] is not
+     * modified by this method, so the caller needs to take care of clearing the
+     * value if it is sensitive.
+     */
+    private static byte[] toUtf8Bytes(char[] chars) {
+        final CharBuffer charBuffer = CharBuffer.wrap(chars);
+        final ByteBuffer byteBuffer = StandardCharsets.UTF_8.encode(charBuffer);
+        final byte[] bytes = Arrays.copyOfRange(byteBuffer.array(), byteBuffer.position(), byteBuffer.limit());
+        Arrays.fill(byteBuffer.array(), (byte) 0); // clear sensitive data
+        return bytes;
+    }
```

Comment:
Copied from: https://github.com/elastic/elasticsearch/blob/113c1916eeffb8fb453d56c97942eefa19bcb087/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/CharArrays.java#L18

Alternatively, we could pull the `CharArrays` utils class out of `x-pack`. Yet this is the only use case in `server`. Moreover, `CharArrays` brings a methods that is narrowed to the security scope, namely `constantTimeEquals`.
Another alternative would be to have the `utf8BytesToChars` and `toUtf8Bytes` as static members of `SecureString` since right now uses are tied to `SecureString`.

All in all, I think copying them here for this sole use is the best alternative.